src/functional/ffront/func_to_foast.py

```diff
@@ -93,8 +94,21 @@ def _preprocess_definition_ast(cls, definition_ast: ast.AST) -> ast.AST:
         return ucc
 
     @classmethod
-    def _postprocess_dialect_ast(cls, dialect_ast: foast.FieldOperator) -> foast.FieldOperator:
-        return FieldOperatorTypeDeduction.apply(dialect_ast)
+    def _postprocess_dialect_ast(
+        cls, dialect_ast: foast.FieldOperator, captured_vars: CapturedVars
+    ) -> foast.FieldOperator:
+        # ensure annotated matches with deduced return type
+        annotated_return_type = symbol_makers.make_symbol_type_from_typing(
+            captured_vars.annotations["return"]
+        )
+        # TODO(tehrengruber): use `type_info.return_type` when the type of the
+        #  arguments becomes available here
+        foast_type = FieldOperatorTypeDeduction.apply(dialect_ast)
```

Comment:
This is a seperate step that should not be intermangled with the return type check, but done seperately as suggested above. Just to be sure: This is not a type in any way. It is a typed ast.
```suggestion
```